### DISCUSSION TITLE
Problem with Tasmota smart meter and script

### DISCUSSION THREAD

**Discussion by joachim-beck** [arendst/Tasmota#17993] [Q&A]
Hello,
I am using a Hichi IR wifi smart meter head with precompiled Tasmota and SML scripting.

My smart meter is an ITRON EM-214 (OBIS ASCII) for which i wrote a script. The script works and delivers correct values with on exception:

The smart meter delivers among others the values with decoder string 1.7.0 and three values 31.7.0, 51.7.0 und 71.7.0. See script below.

The measurements for 31.7.0, 51.7.0 and 71.7.0 are always correct. the measurement 1.7.0 is sometimes correct but contains mostly the value of one the measurements 31.7.0, 51.7.0, 71.7.0. It looks like that the decoder detects 1.7.0 inside the 31.7.0, 51.7.0 or 71.7.0.

Is this a bug in my configuration or a bug in the software?

Here is the data which the smart meter sends:

---------------------------------------------------

/ITR5EM214

[STX]F.F(00000000)

0.0.0(1ITR0054844627)

0.2.0(V01.69#EEA3546C)

0.2.1(00017279)

1.7.0(00.51*kW)

1.8.0(019832.8*kWh)

2.7.0(00.48*kW)

2.8.0(021324.2*kWh)

31.7.0(-002.0*A)

51.7.0(-002.0*A)

71.7.0(+003.0*A)

C.1(54844627)

C.5.0(L1-L2-L3)

C.13.21(00000000*secs)

C.13.22(00000000)

C.90(00000000)

!

----------------------------------------
and here is my script:

----------------------------------------
>D

>B

=>sensor53 r

>M 1

+1,3,o,0,300,EM214,1,30,2F3F210D0A,063030300D0A

1,\n1.8.0(@1,Gesamtbezug,kWh,Gesamtbezug,1

1,2.8.0(@1,Gesamteinspeisung,kWh,Gesamteinspeisung,1

1,1.7.0(@1,Momentanbezug,kW,Momentanbezug,2                                      <<< this value contains often the wrong value

1,2.7.0(@1,Momentaneinspeisung,kW,Momentaneinspeisung,2

1,31.7.0(@1,Momentanstrom-L1,A,Momentanstrom-L1,1                             <<< from here

1,51.7.0(@1,Momentanstrom-L2,A,Momentanstrom-L2,1                             <<< or here

1,71.7.0(@1,Momentanstrom-L3,A,Momentanstrom-L3,1                             <<< or here

#

Thank you very much for any help with this issue.

Best regards

Joachim

**Comment by gemu2015** [comment]
switch to line compare instead of pattern compare (this will be the default in future versions)
by adding this line (set special option)
1,=so2,2

**Comment by joachim-beck** [comment]
Hello,

I tried this, but it changes nothing. The value 1.7.0 is still wrong most of time.
Maybe I have added the option in a wrong way.
Here the actual script:

>D
>B
=>sensor53 r
>M 1
+1,3,o,0,300,EM214,1,30,2F3F210D0A,063030300D0A
1,=so2,2
1,1.8.0(@1,Gesamtbezug,kWh,Gesamtbezug,1
1,2.8.0(@1,Gesamteinspeisung,kWh,Gesamteinspeisung,1
1,1.7.0(@1,Momentanbezug,kW,Momentanbezug,2
1,2.7.0(@1,Momentaneinspeisung,kW,Momentaneinspeisung,2
1,31.7.0(@1,Momentanstrom-L1,A,Momentanstrom-L1,1
1,51.7.0(@1,Momentanstrom-L2,A,Momentanstrom-L2,1
1,71.7.0(@1,Momentanstrom-L3,A,Momentanstrom-L3,1
#

Thank you for your help!
Joachim

**Reply by gemu2015** [reply]
You should use the latest dev version
This option was added recently

**Reply by CMMenzel** [reply]
Hi @joachim-beck - did you ever resolve this issue?
I still occasionally get wrong readings in 1.7.0 - often negative values that seem to spill over from 31.7.0.
Would be great to hear if and how you have solved it :-) Thanks

**Comment by CMMenzel** [comment]
> Hello, I am using a Hichi IR wifi smart meter head with precompiled Tasmota and SML scripting.
> 
> My smart meter is an ITRON EM-214 (OBIS ASCII) for which i wrote a script. The script works and delivers correct values with on exception:
> 
> The smart meter delivers among others the values with decoder string 1.7.0 and three values 31.7.0, 51.7.0 und 71.7.0. See script below.
> 
> The measurements for 31.7.0, 51.7.0 and 71.7.0 are always correct. the measurement 1.7.0 is sometimes correct but contains mostly the value of one the measurements 31.7.0, 51.7.0, 71.7.0. It looks like that the decoder detects 1.7.0 inside the 31.7.0, 51.7.0 or 71.7.0.
> 
> Is this a bug in my configuration or a bug in the software?
> 
> Here is the data which the smart meter sends:
> 
> /ITR5EM214
> 
> [STX]F.F(00000000)
> 
> 0.0.0(1ITR0054844627)
> 
> 0.2.0(V01.69#EEA3546C)
> 
> 0.2.1(00017279)
> 
> 1.7.0(00.51*kW)
> 
> 1.8.0(019832.8*kWh)
> 
> 2.7.0(00.48*kW)
> 
> 2.8.0(021324.2*kWh)
> 
> 31.7.0(-002.0*A)
> 
> 51.7.0(-002.0*A)
> 
> 71.7.0(+003.0*A)
> 
> C.1(54844627)
> 
> C.5.0(L1-L2-L3)
> 
> C.13.21(00000000*secs)
> 
> C.13.22(00000000)
> 
> C.90(00000000)
> 
> !
> 
> and here is my script:
> 
> > D
> 
> > B
> 
> =>sensor53 r
> 
> > M 1
> 
> +1,3,o,0,300,EM214,1,30,2F3F210D0A,063030300D0A
> 
> 1,\n1.8.0(@1,Gesamtbezug,kWh,Gesamtbezug,1
> 
> 1,2.8.0(@1,Gesamteinspeisung,kWh,Gesamteinspeisung,1
> 
> 1,1.7.0(@1,Momentanbezug,kW,Momentanbezug,2 <<< this value contains often the wrong value
> 
> 1,2.7.0(@1,Momentaneinspeisung,kW,Momentaneinspeisung,2
> 
> 1,31.7.0(@1,Momentanstrom-L1,A,Momentanstrom-L1,1 <<< from here
> 
> 1,51.7.0(@1,Momentanstrom-L2,A,Momentanstrom-L2,1 <<< or here
> 
> 1,71.7.0(@1,Momentanstrom-L3,A,Momentanstrom-L3,1 <<< or here
> 
> Thank you very much for any help with this issue.
> 
> Best regards
> 
> Joachim

Hi Joachim,

So glad I found your SML script as I could not get my EM214 meter to work.. 

Did you find a solution to the problem with 1.7.0?


Many thanks
Chris

**Reply by eduard-bs** [reply]
I also used the Skirpt and the value 2.8.0 was sometimes correct and sometimes not.

I increased the time interval from 30 to 40 and then it worked stably for me.

regards
Eduard
